MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/MultipleHeatmapStylingActivity.java

```diff
@@ -0,0 +1,390 @@
+      .target(new LatLng(34.056684, -118.254002))
+      .zoom(11.047)
+      .build();
+    mapboxMap.animateCamera(
+      CameraUpdateFactory.newCameraPosition(cameraPositionForFragmentMap), 2600);
+    addHeatmapDataSource();
+    initHeatmapColors();
+    initHeatmapRadiusStops();
+    initHeatmapIntensityStops();
+    addHeatmapLayer();
+    findViewById(R.id.switch_heatmap_style_fab).setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+        if (index == listOfHeatmapColors.length - 1) {
+          index = 0;
+        }
+        if (firstLoad) {
+          index = 1;
+          firstLoad = false;
+        }
+        mapboxMap.getLayer(HEATMAP_LAYER_ID).setProperties(
+          heatmapColor(listOfHeatmapColors[index]),
+          heatmapRadius(listOfHeatmapRadiusStops[index]),
+          heatmapIntensity(listOfHeatmapIntensityStops[index])
+        );
+        index++;
+      }
+    });
+  }
+
+  private void addHeatmapDataSource() {
+    mapboxMap.addSource(new GeoJsonSource(HEATMAP_SOURCE_ID,
+      loadGeoJsonFromAsset("la_heatmap_styling_points.geojson")));
+  }
+
+  private void addHeatmapLayer() {
+    // Create the heatmap layer
+    HeatmapLayer layer = new HeatmapLayer(HEATMAP_LAYER_ID, HEATMAP_SOURCE_ID);
+
+    // Heatmap layer disappears at whatever zoom level is set as the maximum
+    layer.setMaxZoom(18);
+
+    layer.setProperties(
+      // Color ramp for heatmap.  Domain is 0 (low) to 1 (high).
+      // Begin color ramp at 0-stop with a 0-transparency color to create a blur-like effect.
+      heatmapColor(listOfHeatmapColors[index]),
+
+      // Increase the heatmap color weight weight by zoom level
+      // heatmap-intensity is a multiplier on top of heatmap-weight
+      heatmapIntensity(listOfHeatmapIntensityStops[index]),
+
+      // Adjust the heatmap radius by zoom level
+      heatmapRadius(listOfHeatmapRadiusStops[index]
+      ),
+
+      heatmapOpacity(1f)
+    );
+
+    // Add the heatmap layer to the map and above the "water-label" layer
+    mapboxMap.addLayerAbove(layer, "waterway-label");
+  }
+
+  @Override
+  protected void onStart() {
+    super.onStart();
+    mapView.onStart();
+  }
+
+  @Override
+  protected void onResume() {
+    super.onResume();
+    mapView.onResume();
+  }
+
+  @Override
+  protected void onPause() {
+    super.onPause();
+    mapView.onPause();
+  }
+
+  @Override
+  protected void onStop() {
+    super.onStop();
+    mapView.onStop();
+  }
+
+  @Override
+  public void onSaveInstanceState(Bundle outState) {
+    super.onSaveInstanceState(outState);
+    mapView.onSaveInstanceState(outState);
+  }
+
+  @Override
+  public void onLowMemory() {
+    super.onLowMemory();
+    mapView.onLowMemory();
+  }
+
+  @Override
+  public void onDestroy() {
+    super.onDestroy();
+    mapView.onDestroy();
+  }
+
+  private void initHeatmapColors() {
+    listOfHeatmapColors = new Expression[] {
```

Comment:
You numbered some of these but not all of them. Make it consistent?

Reply:
Added numbers for all arrays